test/integration/services/ledger/helpers.ts

```diff
@@ -235,6 +235,88 @@ export const recordOnChainIntraLedgerPayment = async ({
   })
 }
 
+export const recordLnTradeIntraAccountTxn = async ({
+  senderWalletDescriptor,
+  recipientWalletDescriptor,
+  paymentAmount,
+}) => {
+  const { metadata, debitAccountAdditionalMetadata } =
+    LedgerFacade.LnTradeIntraAccountLedgerMetadata({
+      paymentHash: crypto.randomUUID() as PaymentHash,
+      amountDisplayCurrency: Number(
+        paymentAmount.usd.amount,
+      ) as DisplayCurrencyBaseAmount,
+      feeDisplayCurrency: 0 as DisplayCurrencyBaseAmount,
+      displayCurrency: DisplayCurrency.Usd,
+      pubkey: crypto.randomUUID() as Pubkey,
+      paymentFlow: {
+        btcPaymentAmount: paymentAmount.btc,
+        usdPaymentAmount: paymentAmount.usd,
+        btcProtocolFee: ZERO_SATS,
+        usdProtocolFee: ZERO_CENTS,
+      } as PaymentFlowState<WalletCurrency, WalletCurrency>,
+    })
+
+  return LedgerFacade.recordIntraledger({
+    description: "sends/receives ln trade",
+    amount: paymentAmount,
+    senderWalletDescriptor,
+    recipientWalletDescriptor,
+    metadata,
+    additionalDebitMetadata: debitAccountAdditionalMetadata,
+  })
+}
+
+export const recordWalletIdTradeIntraAccountTxn = async ({
+  senderWalletDescriptor,
+  recipientWalletDescriptor,
+  paymentAmount,
+}) => {
+  const metadata = LedgerFacade.WalletIdTradeIntraAccountLedgerMetadata({
+    amountDisplayCurrency: Number(paymentAmount.usd.amount) as DisplayCurrencyBaseAmount,
+    feeDisplayCurrency: 0 as DisplayCurrencyBaseAmount,
+    displayCurrency: DisplayCurrency.Usd,
+    paymentFlow: {
+      btcPaymentAmount: paymentAmount.btc,
+      usdPaymentAmount: paymentAmount.usd,
+      btcProtocolFee: ZERO_SATS,
+      usdProtocolFee: ZERO_CENTS,
+    } as PaymentFlowState<WalletCurrency, WalletCurrency>,
```

Comment:
```suggestion
  const paymentFlow: PaymentFlowState<WalletCurrency, WalletCurrency> = {
    btcPaymentAmount: paymentAmount.btc,
    usdPaymentAmount: paymentAmount.usd,
    btcProtocolFee: ZERO_SATS,
    usdProtocolFee: ZERO_CENTS,
  }
  const metadata = LedgerFacade.WalletIdTradeIntraAccountLedgerMetadata({
    amountDisplayCurrency: Number(paymentAmount.usd.amount) as DisplayCurrencyBaseAmount,
    feeDisplayCurrency: 0 as DisplayCurrencyBaseAmount,
    displayCurrency: DisplayCurrency.Usd,
    paymentFlow,
  })
```

the type for `paymentFlow` doesn't match, but the catching is hiding the mismatch. is that intended?

Reply:
Yup this is just for these tests as a shortcut to get the exact properties we're interested in. The entire payment flow object is usually passed in when we actually call these in the application.